app/uk/gov/hmrc/apidefinition/services/APIDefinitionService.scala

```diff
@@ -62,6 +67,27 @@ class APIDefinitionService @Inject()(wso2Publisher: WSO2APIPublisher,
     }
   }
 
+  private def checkAPIDefinitionForStatusChanges(apiDefinition: APIDefinition): Unit = {
+    def findStatusDifferences(existingAPIVersions: Seq[APIVersion], newAPIVersions: Seq[APIVersion]): Future[Seq[(String, APIStatus, APIStatus)]] =
+      Future {
+        (existingAPIVersions ++ newAPIVersions)
+          .groupBy(_.version)
+          .filter(v => v._2.size == 2)
+          .filterNot(v => v._2.head.status == v._2.last.status)
+          .map(v => (v._1, v._2.head.status, v._2.last.status))
+          .toSeq
+      }
+
+    for {
+      apiDefinitionOption <- apiDefinitionRepository.fetchByName(apiDefinition.name)
+      differences <- {
+        if (apiDefinitionOption.isDefined) findStatusDifferences(apiDefinitionOption.get.versions, apiDefinition.versions)
+        else findStatusDifferences(Seq.empty, apiDefinition.versions)
```

Comment:
I think we'd never get any differences when it gets into the "else" branch. So we could just do the findStatusDifferences inside apiDefinitionOption.map